static_frame/core/bus.py

```diff
@@ -768,9 +769,37 @@ def relabel(
         Args:
             index: {relabel_input_index}
         """
-        # NOTE: can be done without going trhough a series
+        if self._store is not None and not self._loaded_all:
+            own_index, index_init = relabel_index(
+                relabel=index,
+                original=self._index,
+                index_constructor=None,
+            )
+
+            relabeled = self.__class__(
+                None,  # will generate FrameDeferred array
+                index=index_init,  # type: ignore
+                index_constructor=index_constructor,
+                store=self._store.__copy__(),
+                config=self._config,
+                max_persist=self._max_persist,
+                own_index=own_index,
+            )
+
+            # Match the persistent status
+            relabeled._values_mutable = self._values_mutable.copy()
+            relabeled._loaded = self._loaded.copy()
+            relabeled._loaded_all = self._loaded_all
+
+            if self._max_persist is not None:
+                relabeled._last_loaded = {
+                    relabeled.index[self._index.loc_to_iloc(self_label)]: None
+                    for self_label in self._last_loaded.keys()
+                }
```

Comment:
This is a little sketchy, not because it is incorrect, but because we are manipulating a bundle of private state far from the implementation. I wonder if it is better to implement this as a private method on the `Bus`... also, are there other places where we might do this kind of move that we have not been? If there are, having a dedicated method makes even more sense.

This will also need a bit of testing to make sure we have covered all the relevant cases; as you know, `Bus` persistence is tricky.

Also, should this be `indexi_init` used here and not `self._index`?
```
                relabeled._last_loaded = {
                    relabeled.index[index_init.loc_to_iloc(self_label)]: None
                    for self_label in self._last_loaded.keys()
                }
```